lib/buffer.js

```diff
@@ -832,8 +832,9 @@ Buffer.prototype.writeUIntLE = function(value, offset, byteLength, noAssert) {
   value = +value;
   offset = offset >>> 0;
   byteLength = byteLength >>> 0;
+  var maxBytes = Math.pow(2, 8 * byteLength) - 1;
   if (!noAssert)
-    checkInt(this, value, offset, byteLength, Math.pow(2, 8 * byteLength), 0);
+    checkInt(this, value, offset, byteLength, maxBytes, 0);
```

Comment:
oh whoops. I just moved it b/c it didn't fit in 80 chars, but brain farted that i moved it outside the if. will fix.